crates/ruff_python_ast/src/nodes.rs

```diff
@@ -2578,13 +2577,235 @@ impl Ranged for Identifier {
+    /// Return the [`Int`] as an i8, if it can be represented as that data type.
+    pub fn as_i8(&self) -> Option<i8> {
+        match &self.0 {
+            Number::Small(small) => i8::try_from(*small).ok(),
+            Number::Big(_) => None,
+        }
+    }
+
+    /// Return the [`Int`] as an i16, if it can be represented as that data type.
+    pub fn as_i16(&self) -> Option<i16> {
+        match &self.0 {
+            Number::Small(small) => i16::try_from(*small).ok(),
+            Number::Big(_) => None,
+        }
+    }
+
+    /// Return the [`Int`] as an i32, if it can be represented as that data type.
+    pub fn as_i32(&self) -> Option<i32> {
+        match &self.0 {
+            Number::Small(small) => i32::try_from(*small).ok(),
+            Number::Big(_) => None,
+        }
+    }
+
+    /// Return the [`Int`] as an i64, if it can be represented as that data type.
+    pub fn as_i64(&self) -> Option<i64> {
+        match &self.0 {
+            Number::Small(small) => Some(*small),
+            Number::Big(_) => None,
+        }
+    }
+}
+
+impl std::fmt::Display for Int {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl Debug for Int {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        std::fmt::Display::fmt(self, f)
+    }
+}
+
+impl PartialEq<u8> for Int {
+    fn eq(&self, other: &u8) -> bool {
+        self.as_u8() == Some(*other)
+    }
+}
+
+impl PartialEq<u16> for Int {
+    fn eq(&self, other: &u16) -> bool {
+        self.as_u16() == Some(*other)
+    }
+}
+
+impl PartialEq<u32> for Int {
+    fn eq(&self, other: &u32) -> bool {
+        self.as_u32() == Some(*other)
+    }
+}
+
+impl PartialEq<i8> for Int {
+    fn eq(&self, other: &i8) -> bool {
+        self.as_i8() == Some(*other)
+    }
+}
+
+impl PartialEq<i16> for Int {
+    fn eq(&self, other: &i16) -> bool {
+        self.as_i16() == Some(*other)
+    }
+}
+
+impl PartialEq<i32> for Int {
+    fn eq(&self, other: &i32) -> bool {
+        self.as_i32() == Some(*other)
+    }
+}
+
+impl PartialEq<i64> for Int {
+    fn eq(&self, other: &i64) -> bool {
+        self.as_i64() == Some(*other)
+    }
+}
+
+impl From<u8> for Int {
+    fn from(value: u8) -> Self {
+        Self::small(i64::from(value))
+    }
+}
+
+impl From<u16> for Int {
+    fn from(value: u16) -> Self {
+        Self::small(i64::from(value))
+    }
+}
+
+impl From<u32> for Int {
+    fn from(value: u32) -> Self {
+        Self::small(i64::from(value))
+    }
+}
+
+impl From<i8> for Int {
+    fn from(value: i8) -> Self {
+        Self::small(i64::from(value))
+    }
+}
+
+impl From<i16> for Int {
+    fn from(value: i16) -> Self {
+        Self::small(i64::from(value))
+    }
+}
+
+impl From<i32> for Int {
+    fn from(value: i32) -> Self {
+        Self::small(i64::from(value))
+    }
+}
+
+impl From<i64> for Int {
+    fn from(value: i64) -> Self {
+        Self::small(value)
+    }
+}
```

Comment:
@MichaReiser - I don't know if you had anything fancier than this in mind. I just did `From`, `as_*`, and `PartialEq` for `u8`, `u16`, `u32`, `i8`, `i16`, `i32`, and `i64`.